pkg/alerts/muting_rules.go

```diff
@@ -35,23 +42,113 @@ type MutingRuleCondition struct {
 	Values    []string `json:"values"`
 }
 
+// MutingRuleScheduleRepeat represents how frequently a MutingRule's schedule repeats.
+type MutingRuleScheduleRepeat string
+
+// MutingRuleScheduleRepeatTypes are intervals for MutingRulesScheduleRepeat.
+var MutingRuleScheduleRepeatTypes = struct {
+	// DAILY - Schedule repeats once per calendar day.
+	DAILY MutingRuleScheduleRepeat
+	// WEEKLY - Schedule repeats once per specified day per calendar week.
+	WEEKLY MutingRuleScheduleRepeat
+	// MONTHLY - Schedule repeats once per calendar month.
+	MONTHLY MutingRuleScheduleRepeat
+}{
+	DAILY:   "DAILY",
+	WEEKLY:  "WEEKLY",
+	MONTHLY: "MONTHLY",
+}
+
+// DayOfWeek is used to configure a WEEKLY scheduled MutingRule.
+type DayOfWeek string
+
+// DayOfWeekTypes are days of the week for DayOfWeek.
+var DayOfWeekTypes = struct {
+	MONDAY    DayOfWeek
+	TUESDAY   DayOfWeek
+	WEDNESDAY DayOfWeek
+	THURSDAY  DayOfWeek
+	FRIDAY    DayOfWeek
+	SATURDAY  DayOfWeek
+	SUNDAY    DayOfWeek
+}{
+	MONDAY:    "MONDAY",
+	TUESDAY:   "TUESDAY",
+	WEDNESDAY: "WEDNESDAY",
+	THURSDAY:  "THURSDAY",
+	FRIDAY:    "FRIDAY",
+	SATURDAY:  "SATURDAY",
+	SUNDAY:    "SUNDAY",
+}
+
+// NaiveDateTime wraps `time.Time` to remove the time zone offset when JSON marshaling.
+// NaiveDateTime is used for MutingRuleScheduleCreateInput and MutingRuleScheduleUpdateInput fields StartTime, EndTime, and EndRepeat.
+type NaiveDateTime struct {
```

Comment:
Do we also need an `UnmarsalJSON` method for this type?

Also, what do you think about moving this type to the `nrtime` package?  If its only used in nerdgraph by the alerts endpoint, here seems fine, but if its used by other types that we might want to reference in the future, perhaps the `nrtime` package would be a better spot.

Reply:
@zlesnr We shouldn't need an `UnmarsalJSON` for this type because the API returns a normal DateTime.  The muting rules fields that use NaiveDateTime type for the incoming payload are a bit strange because the API actually returns a normal DateTime (with an offset) in the response.  So to be accurate to API, I think we should expose that to users (no unmarshaling).

In response to moving it to the `nrtime` package, I think that's fine.  I know nerdgraph created the type specifically for our use case so I don't know if it's used elsewhere, but I don't think it would hurt to move it to the `nrtime` package in case it is used elsewhere in the future.

Reply:
Okay, that sounds reasonable to leave it in place.